types/sharp/sharp-tests.ts

```diff
@@ -333,17 +333,17 @@ sharp("in.gif", { animated: true }).toFile("out.webp");
 // Type support for limitInputPixels
 sharp({
```

Comment:
I'd love to see some tests added for `withMetadata`, perhaps something along the lines of

```ts
sharp('input.jpg')
  .withMetadata({
    exif: {
      IFD0: {
        Compression: 7,
        PhotometricInterpretation: 2,
        Copyright: '@fdebijl'
        // Etc.
      }
    }
  })
```

It's probably not feasible to test for every exif property, but a handful of the most common ones might be useful to prevent regression. Testing the type parameters might also be a good idea.

Reply:
Hey @Fdebijl, i'm pretty new to Sharp in general, are there any existing tests for metadata that I might build off of? Sorry i'm just doing this work when I can eek out some time for pet projects but would like to get this done as I know how important tests are.

Reply:
I don't think there are any tests along those lines, but my example above should work out of the box. You can just add more metadata properties as needed.